lib/converters/entity_id.js

```diff
@@ -0,0 +1,115 @@
+const entityData = require("./../dataSingleton");
+const pinyin = require("pinyin");
+const translit = require("translit-rus-eng");
+
+const candidates = [
+  function (obj) {
+    // Highest priority: instance-specific name.
+    return (
+      obj &&
+      obj.common &&
+      obj.common.custom &&
+      obj.common.custom[entityData.adapter.namespace] &&
+      obj.common.custom[entityData.adapter.namespace].name
+    );
+  },
+  function (obj) {
+    // Next: common.name, if available.
+    return obj && obj.common && maybeEnglishOrUntranslated(obj.common.name);
+  },
+  function (obj) {
+    // Fallback: ioBroker object ID.
+    return obj && obj._id;
+  },
+];
+
+function maybeEnglishOrUntranslated (hash) {
+  if (typeof hash === "object") {
+    // Entity IDs are always computed from English.
+    return hash.en;
+  }
+
+  return hash;
+};
+
+function entityIdFromObject(obj) {
+  const found = candidates.reduce(function (acc, candidate) {
+    // Poor man's lazy evaluation.
+    if (acc) {
+      return acc;
+    }
+
+    return candidate(obj);
+  }, undefined);
+
+  return handleChineseCharacters(found || "");
+}
+
+// TODO: Inconsistent handling of Chinese here and Russian later in
```

Comment:
Yes, this is inconsistent... TBH, I never really bothered with this code, it was written by @GermanBluefox (I guess). The Russian transliteration was added by someone else as PR. 

Maybe we should move the call to handleChinesesCharacters into replaceInvalidChars as first call or something? Then it would be more consistent and replaceInvalidChars the single place to replace chars (it also replaces German Umlauts).